HYUNSANG_OH/app.js

```diff
@@ -75,45 +80,93 @@ app.get("/userPost/:userId", async (req, res) => {
     `,
     [userId]
   );
-  const result = userPost.map((userPost) => ({
-    ...userPost,
-    postings: JSON.parse(userPost.postings),
-  }));
-  res.status(200).json({ data: result });
+  console.log(userPost);
+  res.status(200).json({ data: userPost });
 });
 
 //회원가입
 app.post("/signup", async (req, res, next) => {
   const { name, email, password, profile_image } = req.body;
+
+  const makeHash = async (password, saltRounds) => {
+    return await bcrypt.hash(password, saltRounds);
+  };
+
+  const hashPassword = await makeHash(password, saltRounds);
+
   await appDataSource.query(
     `
-    INSERT INTO users(
-      user_name,
-      email,
-      user_password,
-      profile_image
-    ) VALUES (?, ?, ?, ?);
+      INSERT INTO users(
+        user_name,
+        email,
+        user_password,
+        profile_image
+      ) VALUES (?, ?, ?, ?);
+      `,
+    [name, email, hashPassword, profile_image]
+  );
+
+  return res.status(201).json({ message: "userCreated" });
+});
+
+//로그인
+app.post("/login", async (req, res) => {
+  const { email, password } = req.body;
+
+  const validity = await appDataSource.query(
```

Comment:
```suggestion
  const [user] = await appDataSource.query(
```
나오는 데이터는 결국 user 데이터일 뿐입니다. 추가로 다음과 같이 작성하면 리스트에서 하나의 요소만 가져올 수 있습니다.